docs/userguide/searchspace.md

```diff
@@ -1,46 +1,235 @@
-# Search spaces
-
-## General information
+# Search Spaces
 
 The term "search space" refers to the domain of possible values for the parameters that are being optimized during a campaign. A search space represents the space within which BayBE explores and searches for the optimal solution. It is implemented via the [`SearchSpace`](baybe.searchspace.core.SearchSpace) class.
 
-A search space might be purely discrete, purely continuous, or hybrid. This is encoded by the ``type`` attribute (see [`SearchSpaceType`](baybe.searchspace.core.SearchSpaceType)). Furthermore, it is possible to restrict certain types of search spaces using [`Constraints`](./constraints.md).
+BayBE offers three different kinds of search spaces, depending on the parameters that are used when defining a search space. A search space can be either __discrete__, __continuous__ or __hybrid__ which is reflected by the [`type`](baybe.searchspace.core.SearchSpaceType) property. BayBE offers different ways of constructing and working with the different kinds of search spaces.
+
+## Discrete search spaces
+
+A search space in which all parameters are discrete, i.e., in which all parameters are [`DiscreteParameter`](baybe.parameters.base.DiscreteParameter), is a **discrete search space**. BayBE offers different ways of constructing a discrete search space.
```

Comment:
one aspect that could be worth mentioning about discerete subspaces: metadata and in particular the ability to use `dont_recommend`

Reply:
Forgot to include. Will change and then ping here.

Reply:
Included a section on metadata in general. Please have a look and resolve if happy